receiver/k8sclusterreceiver/collection/metadata.go

```diff
@@ -25,42 +25,158 @@ import (
 )
 
 const (
-	// Keys for K8s properties
+	// Keys for K8s metadata
 	k8sKeyWorkLoadKind = "k8s.workload.kind"
 	k8sKeyWorkLoadName = "k8s.workload.name"
 )
 
-// KubernetesMetadata currently provides a reasonable way to store metadata
-// and may changes depending on how we decide to handle property updates.
+// KubernetesMetadata associates a resource to a set of properties.
 type KubernetesMetadata struct {
+	// resourceIDKey is the label key of UID label for the resource.
 	resourceIDKey string
-	resourceID    string
-	properties    map[string]string
+	// resourceID is the Kubernetes UID of the resource. In case of
+	// containers, this value is the container id.
+	resourceID string
+	// metadata is a set of key-value pairs that describe a resource.
+	metadata map[string]string
 }
 
 // getGenericMetadata is responsible for collecting metadata from K8s resources that
 // live on v1.ObjectMeta.
 func getGenericMetadata(om *v1.ObjectMeta, resourceType string) *KubernetesMetadata {
 	rType := strings.ToLower(resourceType)
-	properties := utils.MergeStringMaps(map[string]string{}, om.Labels)
+	metadata := utils.MergeStringMaps(map[string]string{}, om.Labels)
 
-	properties[k8sKeyWorkLoadKind] = rType
-	properties[k8sKeyWorkLoadName] = om.Name
-	properties[fmt.Sprintf("%s.creation_timestamp",
+	metadata[k8sKeyWorkLoadKind] = rType
+	metadata[k8sKeyWorkLoadName] = om.Name
+	metadata[fmt.Sprintf("%s.creation_timestamp",
 		rType)] = om.GetCreationTimestamp().Format(time.RFC3339)
 
 	for _, or := range om.OwnerReferences {
-		properties[strings.ToLower(or.Kind)] = or.Name
-		properties[strings.ToLower(or.Kind)+"_uid"] = string(or.UID)
+		metadata[strings.ToLower(or.Kind)] = or.Name
+		metadata[strings.ToLower(or.Kind)+"_uid"] = string(or.UID)
 	}
 
 	return &KubernetesMetadata{
 		resourceIDKey: getResourceIDKey(rType),
 		resourceID:    string(om.UID),
-		properties:    properties,
+		metadata:      metadata,
 	}
 }
 
 func getResourceIDKey(rType string) string {
 	return fmt.Sprintf("k8s.%s.uid", rType)
 }
+
+// mergeKubernetesMetadataMaps merges maps of string (resource id) to
+// KubernetesMetadata into a single map.
+func mergeKubernetesMetadataMaps(maps ...map[string]*KubernetesMetadata) map[string]*KubernetesMetadata {
```

Comment:
Can we define a type alias for the string in `map[string]*KubernetesMetadata` to give it a descriptive name and make it clear that it is the resource id?

Reply:
Added `collection.ResourceID`.